posthog/warehouse/api/available_sources.py

```diff
@@ -0,0 +1,754 @@
+                                                    required=True,
+                                                    placeholder="User1",
+                                                ),
+                                                SourceFieldInputConfig(
+                                                    name="password",
+                                                    label="Tunnel password",
+                                                    type=Type4.PASSWORD,
+                                                    required=True,
+                                                    placeholder="",
+                                                ),
+                                            ],
+                                        ),
+                                    ),
+                                    Option(
+                                        label="Key pair",
+                                        value="keypair",
+                                        fields=cast(
+                                            list[FieldType],
+                                            [
+                                                SourceFieldInputConfig(
+                                                    name="username",
+                                                    label="Tunnel username",
+                                                    type=Type4.TEXT,
+                                                    required=False,
+                                                    placeholder="User1",
+                                                ),
+                                                SourceFieldInputConfig(
+                                                    name="private_key",
+                                                    label="Tunnel private key",
+                                                    type=Type4.TEXTAREA,
+                                                    required=True,
+                                                    placeholder="",
+                                                ),
+                                                SourceFieldInputConfig(
+                                                    name="passphrase",
+                                                    label="Tunnel passphrase",
+                                                    type=Type4.PASSWORD,
+                                                    required=False,
+                                                    placeholder="",
+                                                ),
+                                            ],
+                                        ),
+                                    ),
+                                ],
+                            ),
+                        ],
+                    ),
+                ),
+            ],
+        ),
+    ),
+    ExternalDataSource.Type.MYSQL: SourceConfig(
```

Comment:
I would consider something that allows localizing these configs to the source folder/definition. Doesn't block this PR but a registry pattern like how max ai tools work would be useful in reducing the amount of file traversal for adding a source. In other words, if I add a source, I should just be in a subfolder and all the config happens in that one subfolder as long as I inherit the right class

Reply:
Yuup, that's how it will work - this is just a middle step towards that